src/CMakeLists.txt

```diff
@@ -108,7 +108,7 @@ function(valhalla_module)
   # These options do not affect the ABI and should therefore be used whenever possible for
   # predictable numerical results
   # For the x86-64 compiler, these extensions are enabled by default.
-  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|AMD64|x64|x86|x86_64")
+  if(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|AMD64|x64|x86|x86_64") AND NOT CMAKE_OSX_ARCHITECTURES MATCHES "arm64")
```

Comment:
strange, that seems redundant to me. i guess on ios builds it advertizes itself as x86 through `CMAKE_SYSTEM_RPOCESSOR` variable?

Reply:
`CMAKE_SYSTEM_RPOCESSOR  = x86_64` when we build it with arm64. I will figure out more soon, maybe some of dependencies override it

Reply:
@kevinkreiser , it fixes an issue with new Macs on arm64.

Reply:
`CMAKE_OSX_ARCHITECTURES=arm64`  setup `-arch arm64` flag to compiler. I don't know, why `CMAKE_SYSTEM_PROCESSOR=x86_64`

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/20989

Reply:
oh i see its for the new mac arm architectures, thanks